--- mod_logdb_mysql5.erl.old	2008-10-30 11:57:56.000000000 +0100
+++ mod_logdb_mysql5.erl	2008-10-30 11:58:22.000000000 +0100
@@ -503,7 +503,7 @@
                 SQuery = ["INSERT INTO ",STable," ",
                            "(owner_id,peer_name_id,peer_server_id,at,count) ",
                               "SELECT owner_id,peer_name_id,peer_server_id,\"",Date,"\",count(*) ",
-                                 "FROM ",Table," GROUP BY owner_id,peer_name_id,peer_server_id;"],
+                                 "FROM ",Table," WHERE ext is NULL GROUP BY owner_id,peer_name_id,peer_server_id;"],
 
                 case sql_query_internal(DBRef, SQuery) of
                      {updated, 0} ->
